### PR TITLE
Improve documentation on common macros

### LINKED ISSUE #5070: Improve xref macro explanations in commonly-used-macros document
Thanks @escattone. FWIW I am really not sure about the name. But I do think we should add it to [list of commonly used macros](https://developer.mozilla.org/en-US/docs/MDN/Structures/Macros/Commonly-used_macros). 

It might be useful to generally add more docs - so we currently say "links to a path in the XXXX" but perhaps something like "cssxref links to a page in the CSS Reference (any page with a stub nested below Web/CSS/Reference).

_Originally posted by @hamishwillee in https://github.com/mdn/content/pull/4226#discussion_r632242487_

### PR DESCRIPTION
**Pull request by hamishwillee**
Fixes #5070

The [commonly used macros > linking to pages in reference](https://developer.mozilla.org/en-US/docs/MDN/Structures/Macros/Commonly-used_macros#linking_to_pages_in_references) docs were displayed as a list that was quite hard to parse, and for which some had examples and some did not. It was also missing the (not very useful) event reference.

To make this more useful I have formatted as a table with macro, reference area and examples columns. That makes it very easy to see where these things might be useful, and how they are used. I also added a note to highlight that one benefit is not just the simplicity, but that these are locale aware.

FYI @escattone

### CODE REVIEW

File: files/en-us/mdn/structures/macros/commonly-used_macros/index.html [hunk 1]
```diff
@@ -48,21 +48,86 @@ <h3 id="Creating_a_single_hyperlink">Creating a single hyperlink</h3>
 
 <h3 id="Linking_to_pages_in_references">Linking to pages in references</h3>
 
-<p>There are various macros for linking to pages in specific reference areas of MDN.</p>
+<p>There are macros for locale-independent linking to pages in specific reference areas of MDN: Javascript, CSS, HTML elements, SVG etc.</p>
+
+<p>The macros are easy to use. Minimally all you need to do is specify the name of the item to link to in the first argument. Most macros will also take a second argument allowing you to change the display text (documentation can be found at the links in the left-most column below).</p>
+
+<table class="standard-table">
+    <caption>Compound assignment operators</caption>
```
Comment (Elchi3) [inline files/en-us/mdn/structures/macros/commonly-used_macros/index.html:56]: I think captions are going away in markdown-land but for now we maybe want them still?

```suggestion
    <caption>MDN reference link macros</caption>
```

**Review by Elchi3 — DISMISSED** [review]
Looks _a lot_ better to me! 🎉 Just a nit

**Comment by escattone** [comment]
Looks great to me as well, thank you @hamishwillee! 🎉

File: files/en-us/mdn/structures/macros/commonly-used_macros/index.html (hunk 1 above)
Comment (hamishwillee) [reply files/en-us/mdn/structures/macros/commonly-used_macros/index.html:56]: Thanks - that's a copy paste error from when I copied the page. I will delete.

**Comment by hamishwillee** [comment]
@chrisdavidmills or @Elchi3  Can you please hit approve - I have addressed the nit (and thanks for the review!!!)

**Review by Elchi3 — APPROVED** [review]
